src/promptflow/promptflow/batch/_base_executor_proxy.py

```diff
@@ -39,3 +48,70 @@ async def exec_aggregation_async(
     ) -> AggregationResult:
         """Execute aggregation nodes"""
         raise NotImplementedError()
+
+
+class APIBasedExecutorProxy(AbstractExecutorProxy):
+    @property
+    def api_endpoint(self) -> str:
+        raise NotImplementedError()
+
+    async def exec_line_async(
+        self,
+        inputs: Mapping[str, Any],
+        index: Optional[int] = None,
+        run_id: Optional[str] = None,
+    ) -> LineResult:
+        # ensure service health
+        await self.ensure_executor_health()
+        # call execution api to get line results
+        url = self.api_endpoint + "/Execution"
+        payload = {"run_id": run_id, "line_number": index, "inputs": inputs}
```

Comment:
Better to add log here (maybe every 10 lines for  example) to show which line is executing.

Reply:
Add these logs to batch engine.